File: frontend/src/types/tag.ts
/**
 * 标签类型定义
 *
 * @description 定义标签数据模型的 TypeScript 类型
 */

/**
 * 标签数据模型
 *
 * @description 前端标签数据结构，与后端 Rust 结构体 TagModel 对应
 */
export interface Tag {
  /** 唯一标识符，UUID 格式 */
  id: string;
  /** 标签名称，唯一且不能为空 */
  name: string;
  /** 标签颜色，十六进制格式（如 #3b82f6） */
  color: string;
}

/**
 * 创建标签的参数
 */
export interface CreateTagParams {
  /** 标签名称 */
  name: string;
  /** 标签颜色 */
  color: string;
}

/**
 * 更新标签的参数
 */
export interface UpdateTagParams {
  /** 标签 ID */
  id: string;
  /** 新名称（可选） */
  name?: string;
  /** 新颜色（可选） */
  color?: string;
}

File: src-tauri/src/domain/entities/tag.rs
//! 标签实体模块
//!
//! @description 定义标签(Tag)数据模型的 ORM 映射
//! 标签用于对笔记进行分类和筛选

use sea_orm::entity::prelude::*;

/// 标签实体模型
///
/// # 结构说明
/// 对应数据库中的 `tags` 表
#[derive(Clone, Debug, DeriveEntityModel)]
#[sea_orm(table_name = "tags")]
pub struct Model {
    /// 唯一标识符，UUID 格式
    #[sea_orm(primary_key)]
    pub id: String,

    /// 标签名称，唯一且不能为空
    #[sea_orm(column_name = "name", unique)]
    pub name: String,

    /// 标签颜色，十六进制格式，默认为蓝色
    #[sea_orm(column_name = "color", default_value = "#3b82f6")]
    pub color: String,
}

/// 主动模型行为实现
impl ActiveModelBehavior for ActiveModel {}